dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/MultipleApproverActionlet.java

```diff
@@ -64,95 +54,93 @@ public void executeAction(WorkflowProcessor processor, Map<String, WorkflowActio
-		StringTokenizer st = new StringTokenizer(userIds, ", ");
-		while (st.hasMoreTokens()) {
-			String x = st.nextToken();
+		final Set<User> requiredApprovers = new HashSet<>();
+		final Set<User> hasApproved       = new HashSet<>();
+		final StringTokenizer userIdTokenizer = new StringTokenizer(userIds, StringPool.COMMA);
+		while (userIdTokenizer.hasMoreTokens()) {
 
-			if (Validator.isEmailAddress(x)) {
+			final String userIdToken = userIdTokenizer.nextToken();
+
+			if (Validator.isEmailAddress(userIdToken)) {
 				try {
-					User u = APILocator.getUserAPI().loadByUserByEmail(x, APILocator.getUserAPI().getSystemUser(), false);
 
-					requiredApprovers.add(u);
+					final User user = APILocator.getUserAPI().loadByUserByEmail(userIdToken,
+							APILocator.getUserAPI().getSystemUser(), false);
+
+					requiredApprovers.add(user);
 				} catch (Exception e) {
-					Logger.error(this.getClass(), "Unable to find user with email:" + x);
+
+					Logger.error(this.getClass(), "Unable to find user with email:" + userIdToken);
 				}
 			} else {
 				try {
 
-					User u = APILocator.getUserAPI().loadUserById(x, APILocator.getUserAPI().getSystemUser(), false);
-					requiredApprovers.add(u);
+					final User user = APILocator.getUserAPI().loadUserById(userIdToken,
+							APILocator.getUserAPI().getSystemUser(), false);
+					requiredApprovers.add(user);
 				} catch (Exception e) {
-					Logger.error(this.getClass(), "Unable to find user with userID:" + x);
+					Logger.error(this.getClass(), "Unable to find user with userID:" + userIdToken);
 				}
 
 			}
 		}
+
 		List<WorkflowHistory> histories = processor.getHistory();
 		
 		// add this approval to the history
-		WorkflowHistory h = new WorkflowHistory();
-		h.setActionId(processor.getAction().getId());
-		h.setMadeBy(processor.getUser().getUserId());
-		if(histories == null){
-			histories = new ArrayList<WorkflowHistory>();
-			histories.add(h);
-		}else histories.add(h);
+		final WorkflowHistory workflowHistory = new WorkflowHistory();
+		workflowHistory.setActionId(processor.getAction().getId());
+		workflowHistory.setMadeBy(processor.getUser().getUserId());
+		histories = histories == null?new ArrayList<>():histories;
+		histories.add(workflowHistory);
 		
-		for (User u : requiredApprovers) {
+		for (final User requiredApprover : requiredApprovers) {
+			for (final WorkflowHistory history : histories) {
 
-			for (WorkflowHistory history : histories) {
-				if (history.getActionId().equals(processor.getAction().getId())) {
-					if (u.getUserId().equals(history.getMadeBy())) {
-						hasApproved.add(u);
-					}
+				final Map<String, Object> changeMap = history.getChangeMap();
+				if (history.getActionId().equals(processor.getAction().getId()) && // if it is the action id and it is not reset.
+						!WorkflowHistoryState.RESET.name().equals(changeMap.get("state"))) {
 
-				}
+					if (requiredApprover.getUserId().equals(history.getMadeBy())) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4904110)